filebeat/input/kafka/input.go

```diff
@@ -266,19 +270,27 @@ func (h *groupHandler) createEvent(
 	if versionOk && version.IsAtLeast(sarama.V0_11_0_0) {
 		kafkaFields["headers"] = arrayForKafkaHeaders(message.Headers)
 	}
-	event := beat.Event{
-		Timestamp: timestamp,
-		Fields: common.MapStr{
-			"message": string(message.Value),
-			"kafka":   kafkaFields,
-		},
-		Private: eventMeta{
-			handler: h,
-			message: message,
-		},
-	}
 
-	return event
+	// if azure input, then a check for the message is done regarding the list of events
+
+	var events []beat.Event
+	messages := h.parseMultipleMessages(message.Value)
+	for _, msg := range messages {
+		event := beat.Event{
+			Timestamp: timestamp,
+			Fields: common.MapStr{
+				"message": msg,
```

Comment:
Does this work ok? My recollection is that the explicit string conversion (`string(message.Value)` in the old code) was necessary for a lot of things to work, since otherwise it got interpreted as raw bytes by the backend, which messed up the logs and the indexing.

Reply:
I had no issues with it so far, let me know if you know a more secure option, happy to change it